sympy/vector/tests/test_coordsysrect.py

```diff
@@ -482,3 +482,23 @@ def test_translation_trans_equations():
            (d.x + 1, d.y + 1, d.z)
     assert d._translation_trans_equations(a, d._origin, d.base_scalars(), inverse=True) ==\
            (d.x - 1, d.y - 1, d.z)
+
+
+def test_transformation_functions():
+    C = CoordSys3D("C")
+    D = C.locate_new("D", 3 * C.i)
+    assert D.transformation_from_parent() == (C.x + 3, C.y, C.z)
+    assert D.transformation_to_parent() == (D.x - 3, D.y, D.z)
+    assert express(C.x, D, variables=True) == D.x + 3
+    assert express(D.x, C, variables=True) == C.x - 3
+    E = C.orient_new_axis('E', q1, C.k)
+    assert E.transformation_from_parent() == (C.x*cos(q1) - C.y*sin(q1), C.x*sin(q1) + C.y*cos(q1), C.z)
+    assert E.transformation_to_parent() == (E.x*cos(q1) + E.y*sin(q1), -E.x*sin(q1) + E.y*cos(q1), E.z)
+    assert express(C.x, E, variables=True) == E.x*cos(q1) - E.y*sin(q1)
+    assert express(E.x, C, variables=True) == C.x*cos(q1) + C.y*sin(q1)
+    F = C.create_new("D", lambda x, y, z: (x**3, y, z))
+    F._calculate_inv_trans_equations()
+    assert F.transformation_from_parent() == (C.x**(1/3), C.y, C.z)
```

Comment:
1/3 needs to be S(1)/3, otherwise it won't be a rational.

Reply:
By the way, shouldn't it be the opposite direction?

Reply:
Try to also add one test with `lambda x, y, z: x+3, y, z`. This one should behave exactly like the locate_new one.

Reply:
> By the way, shouldn't it be the opposite direction?
 
I think that direction is OK. Transformation equations defines transformation *from* parent

Reply:
*to* parent should be inverse transformatin. I think that we need to exchange names in `transformation_to_parent` `transformation_from_parent`

Reply:
The transformation defines `F.x == C.x**3`, right? _C_ is the parent so `F.x = C.x**3` is from parent, while `C.x = F.x**(S(1)/3)` is from parent (by inferring the equation). Right?

Reply:
> The transformation defines F.x == C.x**3, right?

I'd say that it should be C.x == F.x**3.
https://en.wikipedia.org/wiki/Spherical_coordinate_system#Cartesian_coordinates
On the LHS of transformation equations we have variables from older coordinate system, in that case from parent.
On the LHS of inverse transformation equations we have variables from newer coordinate system.

Reply:
Well... we agreed that `transformation` applies to the coordinate system, right?
I would expect for `spherical` to have:
```python
>>> G = C.create_new("G", "spherical", variable_names=["r", "theta", "phi"])
>>> G.transform_from_parent()
(sqrt(C.x**2 + C.y**2 + C.z**2), ... )
>>> G.transform_to_parent()
(G.r*sin(G.theta)*cos(G.phi), ... )
```
If we want to specify a passive transformation, we should use:
```python
>>> G = C.create_new("G", lambda x, y, z: (sqrt(x**2 + y**2 + z**2), ... , ... ))
```
I believe we have so far done the exact opposite.

That means, if we want to have a spherical transformation and keep a passive transformation convention (not active), we should give the inverse transformation to spherical to `transform` to get a spherical transformation.

`transformation` may have various forms:

- rotation matrix + tranlsation vector, 
- lambda,
- equation mapping,
- string.

They should all be either passive or active transformation definitions (let's avoid mixing the two).

Given that the previous code uses passive transformations, we should keep the passive convention, i.e. the transformation specifies how the coordinate system gets transformed, while the vectors remain at the same position.